src/rabbit_routing_util.erl

```diff
@@ -212,8 +212,8 @@ queue_declare_method(#'queue.declare'{} = Method, Type, Params) ->
             Method2#'queue.declare'{queue = SQNG()};
         {exchange, SQNG} when is_function(SQNG) ->
             Method2#'queue.declare'{queue = SQNG()};
-        {'reply-queue', SQNG} when is_function(SQNG) ->
```

Comment:
Is this really meant to be commented out?

Reply:
In `ensure_endpoint` the `queue_declare_method` function isn't called for `reply-queue`: https://github.com/rabbitmq/rabbitmq-erlang-client/blob/d741a29f3f28c9fe3f176a4405f28c50871212ac/src/rabbit_routing_util.erl#L128

And the function isn't exported.

Reply:
Our team deletes code we assume to be unused. Those functions are used by plugins such as STOMP, MQTT, AMQP 1.0, Shovel, possibly Federation and their Web-* counterparts. I'd delete it and run the tests of all those.

Reply:
These 2 lines for `reply-queue` are unreachable because the 3 calls to `queue_declare_method` have the `Type` values hardcoded and none of the values are `'reply-queue'`. Open `rabbit_routing_util` and search for `queue_declare_method`.

It's a local function so there are no other calls, and even then there doesn't seem to be much with `reply-queue` in all of RabbitMQ, only `rabbitmq-stomp` has some tests about it and they still pass with this commented out.

It seems to have been added by you in 2d396a63bab9147f9a95e43626947a55a3514675 but it doesn't seem like those two lines were ever reachable. On that commit it's only called for `exchange`, `topic` and `queue`, like today.

I can of course remove the lines instead of commenting but I wanted to get a reaction because it looks like legitimate code but somehow isn't called.